### DISCUSSION TITLE
Prevent slide being draggable on some specific elements by a classname or attribute

### DISCUSSION THREAD

**Discussion by huannn1** [davidjerleke/embla-carousel#581] [Help] [answered]
From my client's requirement, it would be great if an slide that contains input elements and we can prevent dragging slide on these inputs.
Maybe an attribute like is-draggable="false" or by a classname on the element or ?
Thank you

**Comment by davidjerleke** [comment, marked as answer]
Hi @huannn1,

**Inputs shouldn't** trigger any drag interaction. Here's a list of all  [excluded elements by default](https://github.com/davidjerleke/embla-carousel/blob/master/packages/embla-carousel/src/components/DragHandler.ts#L61) that won't trigger drag interactions. As you can see here, the drag interaction [is cancelled](https://github.com/davidjerleke/embla-carousel/blob/master/packages/embla-carousel/src/components/DragHandler.ts#L136) for these elements.

However, if you want to add custom elements, the way to do it is to use the [`watchDrag`](https://www.embla-carousel.com/api/options/#watchdrag) option like so:

```ts
const options = {
  watchDrag: (emblaApi, event) => {
    // do your thing here
   
   // return true if Embla should start dragging
   return true
   
   // return false if Embla should cancel dragging
   return false
  }
}
```

🚨 Please note that you need `v8.0.0-rc14` for this to work.

Best,
David

**Reply by plrthink** [reply]
Well, I want the offset effect, which I could drag the inputs element to trigger the slide, how I can customize this?

**Reply by davidjerleke** [reply]
@plrthink thanks for your question.I have no idea what you want to achieve. Please elaborate or share examples.

**Comment by huannn1** [comment]
It works with default HTML input, in my case it is an rich text input. I'll use watchDrag to cancel drag interaction as your response.

Thank you a lot : )
